packages/medusa/src/migrations/1692953518123-drop_money_amount_constraints_for_pricing_module.ts

```diff
@@ -0,0 +1,45 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export class dropMoneyAmountConstraintsForPricingModule1692953518123
+  implements MigrationInterface
+{
+  name = "dropMoneyAmountConstraintsForPricingModule1692953518123"
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `
+        CREATE TABLE IF NOT EXISTS "money_amount_variant"
+        (
+            "money_amount_id" text NOT NULL,
+            "variant_id" text NOT NULL
+        );
+
+        INSERT INTO "money_amount_variant" ("money_amount_id", "variant_id")
```

Comment:
The link module comes with an id as a primary key, we might have to fit that in here to make it module compatible. cc. @carlos-r-l-rodrigues

Reply:
I believe we won't need to create the table.
The `link-modules` migration can auto generate the migration for the defined links and if it is an existing table, it can will update them to create missing fields.

Reply:
Perhaps it would make sense to merge #4909 first then? [This link definition](https://github.com/medusajs/medusa/blob/5653338f2327f79bed3cf81568fbe10bcbb41cf4/packages/link-modules/src/definitions/product-variant-money-amount.ts) is the one that will set it up right?

Reply:
Sorry just saw this. The link modules migration won't come enabled by default with the core, typeorm will still have to manage the schema state in those cases. To make it easier for people to adopt modules in the future, having an explicit migration that they've run in typeorm might be beneficial. wdyt?

Reply:
you're right. this has to be done even if not using modules.

Reply:
also, for existing user the migration is necessary has we will have to migrate existing data to the new schema?